File: 백준/Silver/2579. 계단 오르기/계단 오르기.py
import sys
input = sys.stdin.readline
n = int(input())
stair = [int(input().rstrip()) for i in range(n)]

dp = [0] * (10**4+1)
if n == 1:
    print(stair[0])
elif n == 2:
    print(stair[0]+stair[1])
else:
    dp[0]=stair[0]
    dp[1] = max(stair[1], stair[0]+stair[1])
    dp[2] = max(stair[0]+stair[2], stair[1]+stair[2])

    for i in range(3, n):
        dp[i] = max(dp[i-2]+stair[i], stair[i-1]+stair[i]+dp[i-3])

    print(dp[n-1])

File: 백준/Silver/2579. 계단 오르기/계단 오르기.js
let fs = require("fs");
let input = fs.readFileSync("/dev/stdin").toString().trim().split("\n");
// "/dev/stdin"
// "test/Test.txt"
const n = Number(input[0]);
const scores = [];
for (let i = 1; i <= n; i++) {
  scores.push(Number(input[i]));
}

const f = [
  scores[0],
  Math.max(scores[0], scores[0] + scores[1]),
  Math.max(scores[0] + scores[2], scores[1] + scores[2]),
];

for (let i = 3; i < n; i++) {
  let max = Math.max(
    f[i - 3] + scores[i - 1] + scores[i],
    f[i - 2] + scores[i]
  );

  f[i] = max;
}
console.log(f[n - 1]);